### PR TITLE
feat: implement paypal webhook refunds

### PR DESCRIPTION
**Pull request by aht007**
**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

**Post-merge:**
- [ ] [Backported](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2065367719/Making+a+pull+request+for+a+named+release) to latest and next named releases

### CODE REVIEW

File: commerce_coordinator/apps/commercetools/clients.py [hunk 1]
```diff
@@ -420,102 +456,119 @@ def update_return_payment_state_after_successful_refund(self, order_id: str,
...
+            logger.info(
+                f"[CommercetoolsAPIClient] - Creating refund transaction for payment with ID {payment_id} "
+                f"following successful refund {refund['id']}"
+            )
+            refund = self._preprocess_refund_object(refund, psp)
 
             amount_as_money = CTMoney(
-                cent_amount=stripe_refund.amount,
-                currency_code=stripe_refund.currency.upper()
+                cent_amount=float(refund["amount"]),
```
Comment (shafqatfarhan) [inline commerce_coordinator/apps/commercetools/clients.py:537]:
```suggestion
                cent_amount=int(refund["amount"]),
```

File: commerce_coordinator/apps/commercetools/clients.py [hunk 2]
```diff
@@ -420,102 +456,119 @@ def update_return_payment_state_after_successful_refund(self, order_id: str,
...
-                actions=[add_transaction_action]
+                id=payment_id, version=payment_version, actions=[add_transaction_action]
             )
 
             return returned_payment
         except CommercetoolsError as err:
-            context = f"Unable to create refund payment transaction for "\
-                      f"payment {payment_id} and stripe refund {stripe_refund.id}"
+            context = (
+                f"Unable to create refund payment transaction for " f"payment {payment_id} and refund {refund['id']}"
```
Comment (shafqatfarhan) [inline commerce_coordinator/apps/commercetools/clients.py:558]:
```suggestion
                f"Unable to create refund payment transaction for payment {payment_id}, refund {refund['id']}, and PSP: {psp}. Err: {err}"
```

File: commerce_coordinator/apps/commercetools/clients.py [hunk 3]
```diff
@@ -532,16 +585,18 @@ def update_line_item_transition_state_on_fulfillment(self, order_id: str, order_
 
         from_state_key = self.get_state_by_id(from_state_id).key
 
-        logger.info(f"[CommercetoolsAPIClient] - Transitioning line item state for order with ID {order_id}"
-                    f"from {from_state_key} to {new_state_key}")
+        logger.info(
+            f"[CommercetoolsAPIClient] - Transitioning line item state for order with ID {order_id}"
```
Comment (shafqatfarhan) [inline commerce_coordinator/apps/commercetools/clients.py:589]:
```suggestion
            f"[CommercetoolsAPIClient] - Transitioning line item state for order with ID {order_id} "
```

File: commerce_coordinator/apps/commercetools/clients.py [hunk 4]
```diff
@@ -420,102 +456,119 @@ def update_return_payment_state_after_successful_refund(self, order_id: str,
...
-            stripe_refund: stripe.Refund) -> CTPayment:
+        self, payment_id: str, payment_version: int, refund: Refund, psp=EDX_STRIPE_PAYMENT_INTERFACE_NAME
+    ) -> CTPayment:
         """
         Create Commercetools payment transaction for refund
         Args:
             payment_id (str): Payment ID (UUID)
             payment_version (int): Current version of payment
-            stripe_refund (stripe.Refund): Stripe's refund object
+            refund (stripe.Refund): Refund object
```
Comment (shafqatfarhan) [inline commerce_coordinator/apps/commercetools/clients.py:525]:
```suggestion
            refund (Refund): Refund object
```

File: commerce_coordinator/apps/commercetools/utils.py [hunk 5]
```diff
@@ -198,16 +198,18 @@ def find_refund_transaction(payment: Payment, amount: decimal):
...
     """
     translations = {
         'succeeded': TransactionState.SUCCESS,
+        'completed': TransactionState.SUCCESS,
         'pending': TransactionState.PENDING,
         'failed': TransactionState.FAILURE,
+        'canceled': TransactionState.FAILURE,
     }
-    return translations.get(stripe_refund_status.lower(), stripe_refund_status)
+    return translations.get(refund_status.lower(), refund_status)
```
Comment (shafqatfarhan) [inline commerce_coordinator/apps/commercetools/utils.py:212]:
```suggestion
    return translations.get(refund_status.lower(), TransactionState.SUCCESS)
```

**Review by shafqatfarhan — APPROVED** [review]

**Comment by aht007** [comment]
Failing tests here are related to Paypal receipts which will be resolved in its own branch.

**Pull request merged**